servers/account-data-deleter/src/batchDeleteHandler.spec.ts

```diff
@@ -24,178 +26,221 @@ describe('batchDeleteHandler', () => {
   let sentryStub: jest.SpyInstance;
   let loggerError: jest.SpyInstance;
 
+  const deleteFeatureToggle = {
+    name: config.unleash.flags.deletesDisabled.name,
+    stale: false,
+    type: 'release',
+    project: 'default',
+    variants: [],
+    strategies: [],
+    impressionData: false,
+  };
+
   beforeEach(() => {
     jest.restoreAllMocks();
     scheduleStub = jest
       .spyOn(batchDeleteHandler, 'scheduleNextPoll')
       .mockResolvedValue();
-
     sentryStub = jest.spyOn(Sentry, 'captureException');
     loggerError = jest.spyOn(Logger, 'error');
   });
-
-  it('sends an event when the class is initialized', () => {
-    const eventSpy = jest.spyOn(emitter, 'emit').mockClear();
-    jest.spyOn(BatchDeleteHandler.prototype, 'pollQueue').mockResolvedValue();
-    new BatchDeleteHandler(emitter);
-    expect(eventSpy).toHaveBeenCalledWith('pollBatchDelete');
-  });
-  it('invokes listener when pollBatchDelete event is emitted', async () => {
-    const listenerStub = jest
-      .spyOn(batchDeleteHandler, 'pollQueue')
-      .mockResolvedValue();
-    emitter.emit('pollBatchDelete');
-    expect(listenerStub).toHaveBeenCalledTimes(1);
+  afterAll(() => {
+    mockClient.destroy();
   });
-  it('schedules a poll event after some time if no messages returned', async () => {
-    jest
-      .spyOn(SQSClient.prototype, 'send')
-      .mockImplementation(() => Promise.resolve({ Messages: [] }));
-    await batchDeleteHandler.pollQueue();
-    expect(scheduleStub).toHaveBeenCalledWith(300000);
-    expect(scheduleStub).toHaveBeenCalledTimes(1);
+  describe('with feature flags', () => {
+    describe('killswitch enabled', () => {
+      beforeAll(() => {
+        repo.setToggle(config.unleash.flags.deletesDisabled.name, {
+          ...deleteFeatureToggle,
+          enabled: true,
+        });
+      });
+      it('does not process any messages if kill switch is enabled, and schedules new poll', async () => {
```

Comment:
github kind of mangled this diff. this is really the only substantive change. everything else is just nesting with a little bit of additional setup logic to clear out any feature flags.